mfem/mfem
# Enable HIP backend in libCEED

This PR allows MFEM to use libCEED + HIP. 

_Minimum working example to test:_

- Install libCEED.  To keep the build/linking process as simple as possible, I built libCEED with the `clang` and `clang++` in `/opt/rocm/llvm/bin` (set through `CC` and `CXX` when building libCEED). 

- Build MFEM from this branch with the following options: 
```
MFEM_USE_HIP=YES
MFEM_USE_CEED=YES
CEED_DIR=/path/to/libCEED
```
You can also set `HIP_ARCH` accordingly for your system. (I was using gfx906.)

The previous error with 3D problems has been resolved; all meshes should work with MFEM `ex1` and libCEED + HIP.
The MFEM device string is `ceed-hip`.



 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1714](https://github.com/mfem/mfem/pull/1714) | @nbeams | @tzanio | @YohannDudouit + @camierjs + @barker29 + @artv3 | 09/17/20 | 10/13/20 | 10/18/20 | |
